contracts/fungible-token.aes

```diff
@@ -75,12 +85,14 @@ contract FungibleToken =
     only_owner()
     put(state{ total_supply = state.total_supply + value, balances[account] = balance_of(account) + value })
 
+  // TODO check can burn negative, thus generate tokens?
   public stateful function burn(value: int) =
     require(balance_of(Call.caller) >= value, "The value provided for burning is less than account's balance.")
     put(state{ total_supply = state.total_supply - value, balances[Call.caller] = balance_of(Call.caller) - value })
 
   // INTERNAL FUNCTIONS
 
+  // refactor all require calls in functions
```

Comment:
Meaning to have a separate function for each individual require?

Reply:
I'd do so